File: frontend/src/api/crawler.ts
import api from './index'

export interface CrawlerNewsItem {
  title: string
  url: string
  source: string
  summary: string
  publish_time: string
}

export interface CrawlerListResponse {
  news: CrawlerNewsItem[]
  total: number
}

export interface CrawlerSearchResponse {
  news: CrawlerNewsItem[]
  stats: { sources: Record<string, number>; total: number }
}

export const crawlerApi = {
  async getLatest(): Promise<CrawlerListResponse> {
    const { data } = await api.get('/crawler/latest')
    return data
  },

  async search(keyword: string, platform = 'all'): Promise<CrawlerSearchResponse> {
    const { data } = await api.get('/crawler/search', { params: { keyword, platform } })
    return data
  },
}

File: backend/app/schemas/crawler.py
from pydantic import BaseModel


class CrawlerNewsItem(BaseModel):
    title: str
    url: str
    source: str
    summary: str = "暂无摘要"
    publish_time: str = ""


class CrawlerListResponse(BaseModel):
    news: list[CrawlerNewsItem]
    total: int


class CrawlerSearchResponse(BaseModel):
    news: list[CrawlerNewsItem]
    stats: dict
